pkg/utils/k8sutil.go

```diff
@@ -809,3 +811,79 @@ func addInitContainerAnnotation(dpm *v1beta1.Deployment) error {
 	}
 	return nil
 }
+
+// CreateIngress creates ingress rule for a specific function
+func CreateIngress(client kubernetes.Interface, ingressName, funcName, hostname, ns string) error {
+	if hostname == "" {
+		var err error
+		hostname, err = getLocalHostname(funcName)
+		if err != nil {
+			return err
+		}
+	}
+
+	//TODO: skip annotation. We can add it later
+	//ingressAnnotations := map[string]string{
+	//	"kubernetes.io/ingress.class": "nginx",
+	//	"kubernetes.io/tlsacme":      "true",
+	//}
+
+	ingress := &v1beta1.Ingress{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      ingressName,
+			Namespace: ns,
+			//Annotations: ingressAnnotations,
+		},
+		Spec: v1beta1.IngressSpec{
+			Rules: []v1beta1.IngressRule{
+				{
+					Host: hostname,
+					IngressRuleValue: v1beta1.IngressRuleValue{
+						HTTP: &v1beta1.HTTPIngressRuleValue{
+							Paths: []v1beta1.HTTPIngressPath{
+								{
+									Path: "/",
+									Backend: v1beta1.IngressBackend{
+										ServiceName: funcName,
+										ServicePort: intstr.FromInt(8080),
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+
+	_, err := client.ExtensionsV1beta1().Ingresses(ns).Create(ingress)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// getLocalDomain returns hostname
+func getLocalHostname(funcName string) (string, error) {
+	config, err := buildOutOfClusterConfig()
```

Comment:
`config` should be passed as an argument to `getLocalHostname` (so `getLocalHostname` can be tested / reused in other situations).